mkosi/__init__.py

```diff
@@ -4606,10 +4600,13 @@ def run_verb(args: Args, images: Sequence[Config], *, resources: Path) -> None:
 
     if args.verb == Verb.clean:
         if tools:
-            run_clean(args, tools, resources=resources)
+            run_clean(args, tools)
 
         for config in images:
-            run_clean(args, config, resources=resources)
+            run_clean(args, config)
+
+        if last.output_format != OutputFormat.none:
+            run_clean(args, finalize_default_initrd(last, tools=False, resources=resources))
```

Comment:
@DaanDeMeyer This line is deleting a directory, named `initrd`, in my project. I bisected recent changes to this commit and line. The directory layout of my project looks like:
```
os                                                                                                                                                                                         
├── initrd                                                                                                                                                                                 
│   ├── mkosi.conf                                                                                                                                                                         
├── installer                                                                                                                                                                              
│   └── mkosi.conf                                                                                                                                                                         
└── mkosi.conf
```
The build flow is: build the custom `os/initrd/mkosi.conf` image first, then I build the top level image `os/mkosi.conf` which uses that custom initrd, then I build the `os/installer/mkosi.conf` image.

At the start of the `os/mkosi.conf` image build, `mkosi -f` finds the `initrd` project directy and mistakenly deletes it. (Fortunately I had committed all my changes so I didn't lose any files from my `initrd` directory).